### PR TITLE
fix sundials interface for NON-MPI installations

### PR DESCRIPTION
**Pull request by asartori86**
The `FindSUNDIALS.cmake` did not work with `DEAL_II_WITH_MPI=OFF` because the `SUNDIALS_LIB_PAR` was required. 

This patch fixes the above issue and the interface to properly work even if `DEAL_II_WITH_MPI=OFF`

### CODE REVIEW

File: include/deal.II/sundials/arkode.h [hunk 1]
```diff
@@ -490,8 +493,11 @@ namespace SUNDIALS
      * @param data ARKode configuration data
      * @param mpi_comm MPI communicator
      */
-    ARKode(const AdditionalData &data=AdditionalData(),
-           const MPI_Comm mpi_comm = MPI_COMM_WORLD);
+    ARKode(const AdditionalData &data=AdditionalData()
+#ifdef DEAL_II_WITH_MPI
+                                      , const MPI_Comm mpi_comm = MPI_COMM_WORLD
```
Comment (tjhei) [inline include/deal.II/sundials/arkode.h]: Changing the class interface based on DEAL_II_WITH_MPI is not ideal. I would think we should just ignore that argument if in serial. Can you add a comment in the documentation about the role of the communicator if in serial?

**Comment by luca-heltai** [comment]
Sundials tests work both in serial and mpi. A recent PR broke **compilation** when MPI is off. But tests have always worked both in serial and mpi.

File: include/deal.II/sundials/arkode.h (hunk 1 above)
Comment (luca-heltai) [inline include/deal.II/sundials/arkode.h]: Remove this `#ifdef`. The interface should be the same, and both MPI_Comm  and MPI_COMM_WORLD are defined in `deal.II/base/mpi.h`, which is included, also for non MPI compilation.

File: include/deal.II/sundials/ida.h [hunk 2]
```diff
@@ -565,8 +565,11 @@ namespace SUNDIALS
      * @param data IDA configuration data
      * @param mpi_comm MPI communicator
      */
-    IDA(const AdditionalData &data=AdditionalData(),
-        const MPI_Comm mpi_comm = MPI_COMM_WORLD);
+    IDA(const AdditionalData &data=AdditionalData()
+#ifdef DEAL_II_WITH_MPI
+                                   , const MPI_Comm mpi_comm = MPI_COMM_WORLD
+#endif
```
Comment (luca-heltai) [inline include/deal.II/sundials/ida.h]: same here. Remove the ifdef.

File: source/sundials/arkode.cc [hunk 3]
```diff
@@ -208,10 +208,16 @@ namespace SUNDIALS
   }
 
   template <typename VectorType>
-  ARKode<VectorType>::ARKode(const AdditionalData &data, const MPI_Comm mpi_comm) :
+  ARKode<VectorType>::ARKode(const AdditionalData &data
+#ifdef DEAL_II_WITH_MPI
+                             , const MPI_Comm mpi_comm
+#endif
```
Comment (luca-heltai) [inline source/sundials/arkode.cc]: and here.

File: source/sundials/arkode.cc [hunk 4]
```diff
@@ -208,10 +208,16 @@ namespace SUNDIALS
...
+                            ) :
     data(data),
-    arkode_mem(nullptr),
-    communicator(Utilities::MPI::duplicate_communicator(mpi_comm))
+    arkode_mem(nullptr)
+#ifdef DEAL_II_WITH_MPI
+    ,communicator(Utilities::MPI::duplicate_communicator(mpi_comm))
```
Comment (luca-heltai) [inline source/sundials/arkode.cc]: same.

**Comment by asartori86** [comment]
@tjhei @luca-heltai done. Thanks for your suggestions

**Pull request merged**